### DISCUSSION TITLE
Help Compiling PlayCover with PlayTools commit a6646cc

### DISCUSSION THREAD

**Discussion by jimfrize** [PlayCover/PlayCover#1747] [Q&A] [answered]
Hi all, thanks to the PlayCover team, it looks like an amazing piece of software!
I'm desperate to get my copy of Minecraft Pocket Edition working.

Minecraft crashes on launch and I discovered it was something to do with the PlayChain.
That's when I found this commit: [a6646cc](https://github.com/Ryu-ga/PlayTools/commit/a6646cc600a3dba58c97d6b86380b1750d2ef674#diff-5c1281dc1165078e2442cf2034dedde334b14eb75ad21fe18c4d934b62edacae)

This commit looks like it might fix my crash on launch problem. I'm just struggling to build it on my system.

I saw in the PlayCover build guide that PlayTools is actually fetched by Carthage and changing the PlayTools version is a matter of editing the Cartfile...

I edited the Cartfile to contain: github "PlayCover/PlayTools" "a6646cc"
but Carthage doesn't like that: "Failed to check out repository into ... No object named "a6646cc" exists"

I tried a bunch of other things but can't seem to find a solution, any help would be much appreciated!

**Comment by TheMoonThatRises** [comment, marked as answer]
The Cartfile should contain:

```
github "Ryu-ga/PlayTools" a6646cc600a3dba58c97d6b86380b1750d2ef674
```

as this commit is under Ryu-ga's Github and not PlayCover's and the commit hash should be the complete hash.

**Reply by jimfrize** [reply]
Ok, that makes a lot of sense! Thank you very much for the response. Carthage finds the commit now, unfortunately it looks like the commit has errors and depricated code, so I'll have to look for another solution, but it was worth a shot! Thanks again
